### ISSUE TITLE
Can not compile the simple contract

### ISSUE THREAD

**Issue by xni** [ethereum/solidity#3414]
Hi!

I'm getting an internal error

```
$ truffle compile
Compiling ./contracts/Prescription.sol...

InternalCompilerError:
Compilation failed. See above.
```

for the very simple contract

```
pragma solidity ^0.4.17;

contract Fail {
    struct MedicineBoxDefinition {
        uint medicineId;
        uint useBeforeDate;
    }

   function issue(MedicineBoxDefinition[]) {}
}
```

I suppose there is something related to the array passing to the function.

**Comment by axic** [comment]
Too bad `truffle` is always hiding the messages:

```
Internal compiler error during compilation:
/Users/alex/Projects/solidity/libsolidity/codegen/CompilerUtils.cpp(726): Throw in function void dev::solidity::CompilerUtils::convertType(const dev::solidity::Type &, const dev::solidity::Type &, bool, bool, bool)
Dynamic exception type: boost::exception_detail::clone_impl<dev::solidity::InternalCompilerError>
std::exception::what: 
[dev::tag_comment*] = 
```

**Comment by xni** [comment]
I think explanation about why I was wrong is this:

```
For memory arrays, it cannot be a mapping and has to be an ABI type if it is an argument of a publicly-visible function.
```

**Comment by chriseth** [comment]
The actual bug here is that the recommendation to use the new decoder is not properly given. I would prefer to finish the new decoder.

**Comment by chriseth** [comment]
Fixed by #3549
